### PR TITLE
another try to filter master from CircleCI jobs

### PR DESCRIPTION
**Pull request by Borda**
## What does this PR do?

cc: @zcain117 @awaelchli 

## PR review    
Anyone in the community is free to review the PR once the tests have passed.     
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -160,6 +160,8 @@ workflows:
                 - /^pull\/.*$/
                 # only from canonical repository
                 - /^(?!pull\/).*$/
+              # todo: temporary fix as the "only" above contains master
+              ignore: master
```
Comment (Borda) [inline .circleci/config.yml:164]:
```suggestion
```

**Comment by Borda** [comment]
The only what is bordering me is sync with CircleCI...
the case is the when anyone from @PyTorchLightning/core-contributors creates a PR it is first regular branch in this repo and it does not trigger CircleCI as it does not contain `pull/*` then when such breach is turned to PR CircleCi just takes over the state of the branch and does not reset itself to have the `pull/*` property...
so this may lead that a PR without any extra commit which would do the sate restart may silently ignore TPU testing :/

any thoughts @zcain117 @awaelchli ...

**Comment by Borda** [comment]
the simples way out of the hole is that @williamFalcon will mark the TPU tests as mandatory which is anyway what we want...
these test shall be mandatory:
- GPUs (Drone)
- TPU (CircleCI)

**Comment by zcain117** [comment]
> The only what is bordering me is sync with CircleCI...
> the case is the when anyone from @PyTorchLightning/core-contributors creates a PR it is first regular branch in this repo and it does not trigger CircleCI as it does not contain `pull/*` then when such breach is turned to PR CircleCi just takes over the state of the branch and does not reset itself to have the `pull/*` property...
> so this may lead that a PR without any extra commit which would do the sate restart may silently ignore TPU testing :/
> 
> any thoughts @zcain117 @awaelchli ...

So the problem is only for the first commit and only when the PR is made as a non-forked PR by a core contributor?

**Comment by Borda** [comment]
Well, in fact the problem is filter forked and origin
Beach... The forked repo has pull keys but the original has the original branch name so there is no way to fiter it

**Pull request merged**